pkg/engine/engine_test.go

```diff
@@ -1710,3 +1711,191 @@ func TestWrapAsVariableObject(t *testing.T) {
 		})
 	}
 }
+
+func TestGetClusterAutoscalerAddonFuncMap(t *testing.T) {
```

Comment:
@devigned @mboersma thoughts on this testing pattern for the template/view layer? We have this rather lame pattern (my contribution) established, but its coverage surface area is very superficial:

https://github.com/Azure/aks-engine/blob/master/pkg/engine/template_generator_test.go#L36

It seems like there should be a less onerous way of doing this testing, but the below is my best current attempt at thorough UT traversal of this layer.